test/dlc_tests/sanity/test_safety_check.py

```diff
@@ -205,7 +205,7 @@ def _get_latest_package_version(package):
 @pytest.mark.skipif(not is_dlc_cicd_context(), reason="Skipping test because it is not running in dlc cicd infra")
 @pytest.mark.skipif(
     not (
-        is_mainline_context() or is_safety_test_context() or (is_canary_context() and is_time_for_canary_safety_scan())
+        is_safety_test_context() or (is_canary_context() and is_time_for_canary_safety_scan())
```

Comment:
do we need `is_safety_test_context`?

Reply:
I think we would need it in case we go for `do_build == false` option.